src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -37,7 +37,7 @@ class ServicePrincipalMigrationInfo:
 class StorageCredentialValidationResult:
     name: str
     application_id: str
-    directory_id: str
+    directory_id: str | None
```

Comment:
if it's none - it has to move down and have ` = None` as default

Reply:
fixed